reactor-netty-core/src/main/java/reactor/netty/transport/Transport.java

```diff
@@ -155,6 +156,48 @@ else if (configuration().metricsRecorder != null) {
 		}
 	}
 
+	/**
+	 * Whether to enable metrics to be collected and registered in Micrometer's
+	 * {@link io.micrometer.core.instrument.Metrics#globalRegistry globalRegistry}.
+	 * Applications can separately register their own
+	 * {@link io.micrometer.core.instrument.config.MeterFilter filters}.
+	 * For example, to put an upper bound on the number of tags produced:
+	 * <pre class="code">
+	 * MeterFilter filter = ... ;
+	 * Metrics.globalRegistry.config().meterFilter(MeterFilter.maximumAllowableTags(prefix, 100, filter));
+	 * </pre>
+	 * <p>By default this is not enabled.
+	 *
+	 * @param enable true enables metrics collection; false disables it
+	 * @param prefix a prefix for meters' name
+	 * @param protocol a protocol name for meters' tag
+	 * @return a new {@link Transport} reference
+	 */
+	protected T metrics(boolean enable, String prefix, String protocol) {
+		if (enable) {
+			Objects.requireNonNull(prefix, "prefix");
+			Objects.requireNonNull(protocol, "protocol");
+			if (!Metrics.isInstrumentationAvailable()) {
+				throw new UnsupportedOperationException(
+						"To enable metrics, you must add the dependency `io.micrometer:micrometer-core`" +
+								" to the class path first");
+			}
+			T dup = duplicate();
+			dup.configuration().metricsRecorder(() -> new MicrometerChannelMetricsRecorder(prefix, protocol));
```

Comment:
For generic `Transport`s, this may be OK, but for example `HTTP`, we have a specific integration with Micrometer.

You may want to check 
https://github.com/reactor/reactor-netty/blob/main/reactor-netty-http/src/main/java/reactor/netty/http/server/MicrometerHttpServerMetricsRecorder.java
https://github.com/reactor/reactor-netty/blob/main/reactor-netty-http/src/main/java/reactor/netty/http/client/MicrometerHttpClientMetricsRecorder.java

Also I don't want to extend the scope of the feature request, but if I think in perspective, there might be other requests for customising the default integration with Micrometer. Isn't is better if we consider some kind of `Builder` for configuring this default integration?

Reply:
`Transport` uses this contract for the default integration with Micrometer https://github.com/reactor/reactor-netty/blob/17d3c512f7a4ebff609b1298aa30fd027a8482c2/reactor-netty-core/src/main/java/reactor/netty/transport/TransportConfig.java#L294

Reply:
I have updated the commit to ensure they use their own `TCP/UDP/HTTP` recorders.
As you have mentioned `builder`, I think a more practical way is to make these metrics recorders `public` without `final` currently so users can extend them as they want. Using `builder` needs to consider what properties should be allowed to build, and it can be another ticket if someones else really need it.

Reply:
@JamesChenX Agree. Let's make `MicrometerHttpServerMetricsRecorder` and `MicrometerHttpClientMetricsRecorder` public classes with a corresponding constructor for the `prefix`.
Then the users can utilise:
- `reactor.netty.transport.Transport#metrics(boolean, java.util.function.Supplier<? extends reactor.netty.channel.ChannelMetricsRecorder>)`
- `reactor.netty.http.client.HttpClient#metrics(boolean, java.util.function.Supplier<? extends reactor.netty.channel.ChannelMetricsRecorder>)`
- `reactor.netty.http.client.HttpClient#metrics(boolean, java.util.function.Supplier<? extends reactor.netty.channel.ChannelMetricsRecorder>, java.util.function.Function<java.lang.String,java.lang.String>)`
- `reactor.netty.http.server.HttpServer#metrics(boolean, java.util.function.Supplier<? extends reactor.netty.channel.ChannelMetricsRecorder>)`
- `reactor.netty.http.server.HttpServer#metrics(boolean, java.util.function.Supplier<? extends reactor.netty.channel.ChannelMetricsRecorder>, java.util.function.Function<java.lang.String,java.lang.String>)`

Reply:
I have added the constructor for the `prefix` in `MicrometerHttpServerMetricsRecorder` and `MicrometerHttpClientMetricsRecorder`
https://github.com/reactor/reactor-netty/pull/1776/files#diff-c82d769d5cd42c0cd7042f2294d0c6a10b360abd667551c3610c156c76440f8cR41-R43

Reply:
oops sorry :woman_facepalming: